tests/sparseml/onnx/optim/test_analyzer_model.py

```diff
@@ -139,15 +139,20 @@ def test_mode_analyzer_json():
 
 def _test_model_analyzer(model_path: str, expected_output: str):
     analyzer = ModelAnalyzer(model_path)
+    analyzer_dict = analyzer.dict()
 
     analyzer_from_json = ModelAnalyzer.from_dict(expected_output)
-    analyzer_dict = analyzer.dict()
 
-    return (
-        True
-        if (analyzer_dict == expected_output) and (analyzer == analyzer_from_json)
-        else False
-    )
+    assert len(analyzer_dict["nodes"]) == len(expected_output["nodes"])
+    for node, expected_node in zip(analyzer_dict["nodes"], expected_output["nodes"]):
+        assert sorted(node.keys()) == sorted(expected_node.keys())
```

Comment:
nit: maybe instead of sorting lists, we should turn them into sets for comparison? this expresses better our logic behind the assertion

Reply:
Good call i'll make that change